# Este arquivo define o controlador de esforço puro.

# ur5:
#   joint_state_controller:
#     type: joint_state_controller/JointStateController
#     publish_rate: 100


#   simple_effort_controller: # <--- Mude o nome do controlador AQUI
#     type: "effort_controllers/JointGroupEffortController" # <--- Tipo de controlador alterado
#     joints:
#       - shoulder_pan_joint
#       - shoulder_lift_joint
#       - elbow_joint
#       - wrist_1_joint
#       - wrist_2_joint
#       - wrist_3_joint
#     publish_rate: 100 # Hz
#     gains:
#       shoulder_pan_joint: {p: 4000,  d: 200, i: 1, i_clamp: 1}
#       shoulder_lift_joint: {p: 10000,  d: 200, i: 1, i_clamp: 1}
#       elbow_joint: {p: 2000,  d: 20, i: 1, i_clamp: 1}
#       wrist_1_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
#       wrist_2_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
#       wrist_3_joint: {p: 10,  d: 0.1, i: 0, i_clamp: 1} 


  

ur5: 
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 125

  trajectory_controller: 
    type: effort_controllers/JointTrajectoryController
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    gains:
      shoulder_pan_joint: {p: 4000,  d: 200, i: 1, i_clamp: 1}
      shoulder_lift_joint: {p: 10000,  d: 200, i: 1, i_clamp: 1}
      elbow_joint: {p: 2000,  d: 20, i: 1, i_clamp: 1}
      wrist_1_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
      wrist_2_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
      wrist_3_joint: {p: 10,  d: 0.1, i: 0, i_clamp: 1} 
    state_publish_rate: 125

  # velocity_trajectory_controller:
  #   type: velocity_controllers/JointTrajectoryController # <-- Use este tipo!
  #   joints:
  #     - shoulder_pan_joint
  #     - shoulder_lift_joint
  #     - elbow_joint
  #     - wrist_1_joint
  #     - wrist_2_joint
  #     - wrist_3_joint

  #   # Ganhos PID para converter comandos de velocidade em torques (se a interface for effort)
  #   # Estes ganhos são cruciais e devem ser ajustados para o seu robô!
  #   gains:
  #     shoulder_pan_joint: {p: 4000,  d: 200, i: 1, i_clamp: 1}
  #     shoulder_lift_joint: {p: 10000,  d: 200, i: 1, i_clamp: 1}
  #     elbow_joint: {p: 2000,  d: 20, i: 1, i_clamp: 1}
  #     wrist_1_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
  #     wrist_2_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
  #     wrist_3_joint: {p: 10,  d: 0.1, i: 0, i_clamp: 1}
    
  #   state_publish_rate: 125
  #   action_monitor_rate: 10
  #   stop_trajectory_duration: 0.5
  #   # Outras configurações como 'constraints' podem ser adicionadas aqui




# ur5:
#   # Publish all joint states -----------------------------------
#   joint_state_controller:
#     type: joint_state_controller/JointStateController
#     publish_rate: 125  
  
#   # Position Controllers ---------------------------------------
#   shoulder_pan_joint_position_controller:
#     type: effort_controllers/JointPositionController
#     joint: shoulder_pan_joint
#     pid: {p: 4000,  d: 200, i: 1, i_clamp: 1}
#   shoulder_lift_joint_position_controller:
#     type: effort_controllers/JointPositionController
#     joint: shoulder_lift_joint
#     pid: {p: 10000,  d: 200, i: 1, i_clamp: 1}
#   elbow_joint_position_controller:
#     type: effort_controllers/JointPositionController
#     joint: elbow_joint
#     pid: {p: 2000,  d: 20, i: 1, i_clamp: 1}
#   wrist_1_joint_position_controller:
#     type: effort_controllers/JointPositionController
#     joint: wrist_1_joint
#     pid: {p: 500,  d: 1, i: 1, i_clamp: 1}
#   wrist_2_joint_position_controller:
#     type: effort_controllers/JointPositionController
#     joint: wrist_2_joint
#     pid: {p: 500,  d: 1, i: 1, i_clamp: 1}
#   wrist_3_joint_position_controller:
#     type: effort_controllers/JointPositionController
#     joint: wrist_3_joint
#     pid: {p: 10,  d: 0.1, i: 0, i_clamp: 1} 

#   # Velocity Controllers ---------------------------------------
  # shoulder_pan_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: shoulder_pan_joint
  #   pid: {p: 4000,  d: 200, i: 1, i_clamp: 1}
  # shoulder_lift_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: shoulder_lift_joint
  #   pid: {p: 10000,  d: 200, i: 1, i_clamp: 1}
  # elbow_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: elbow_joint
  #   pid: {p: 2000,  d: 20, i: 1, i_clamp: 1}
  # wrist_1_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: wrist_1_joint
  #   pid: {p: 500,  d: 1, i: 1, i_clamp: 1}
  # wrist_2_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: wrist_2_joint
  #   pid: {p: 500,  d: 1, i: 1, i_clamp: 1}
  # wrist_3_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: wrist_3_joint
  #   pid: {p: 10,  d: 0.1, i: 0, i_clamp: 1} 

  #Effort Controllers ---------------------------------------
  # shoulder_pan_joint_effort_controller:
  #   type: "effort_controllers/JointEffortController"
  #   joint: shoulder_pan_joint
  #   # pid: {p: 4000,  d: 200, i: 1, i_clamp: 1}0
  # shoulder_lift_joint_effort_controller:
  #   type: "effort_controllers/JointEffortController"
  #   joint: shoulder_lift_joint
  #   # pid: {p: 10000,  d: 200, i: 1, i_clamp: 1}
  # elbow_joint_effort_controller:
  #   type: "effort_controllers/JointEffortController"
  #   joint: elbow_joint
  #   # pid: {p: 2000,  d: 20, i: 1, i_clamp: 1}
  # wrist_1_joint_effort_controller:
  #   type: "effort_controllers/JointEffortController"
  #   joint: wrist_1_joint
  #   # pid: {p: 500,  d: 1, i: 1, i_clamp: 1}
  # wrist_2_joint_effort_controller:
  #   type: "effort_controllers/JointEffortController"
  #   joint: wrist_2_joint
  #   # pid: {p: 500,  d: 1, i: 1, i_clamp: 1}
  # wrist_3_joint_effort_controller:
  #   type: "effort_controllers/JointEffortController"
  #   joint: wrist_3_joint
  #   # pid: {p: 10,  d: 0.1, i: 0, i_clamp: 1} 
